File: Gaussian_elimination.py
print('Gaussian Elimination Method')
array = [[1.0, 3.0, -5.0, 2.0], [3.0, 11.0, -9.0, 4.0], [-1.0, 1.0, 6.0, 3.0]]


def display_matrix(arr):
    for i in range(3):
        for j in range(4):
            print(str(arr[i][j]) + '  \t', end='')
            if j == 2:
                print('| \t', end='')
        print()

print('Initial value: ')
display_matrix(array)

first = array[1][0]
second = array[2][0]
for j in range(4):
    array[1][j] = array[1][j] - (array[0][j] / array[0][0]) * first
    array[2][j] = array[2][j] - (array[0][j] / array[0][0]) * second

print('\n\nFirst step: ')
display_matrix(array)

third = array[2][1]
for i in range(1, 4):
    array[2][i] = array[2][i] - ((array[1][i] / array[1][1]) * third)

print('\n\nSecond step: ')
display_matrix(array)

File: GaussianElimination.java
package gaussianelimination;

/**
 *
 * @author noyon
 */
public class GaussianElimination {

    public static void main(String[] args) {
        GaussianElimination ge = new GaussianElimination();
        int array[][] = {{1, 3, -5, 2}, {3, 11, -9, 4}, {-1, 1, 6, 3}};
        System.out.println("Initial value: ");
        ge.display_matrix(array);
        int arrayLength = array.length;
        System.out.println(arrayLength + " ");
        int firstZero = array[1][0];
        int secondZero = array[2][0];
        for (int i = 0; i <= array.length; i++) {
            array[1][i] = array[1][i] - ((array[0][i] / array[0][0]) * firstZero);
            array[2][i] = array[2][i] - ((array[0][i] / array[0][0]) * secondZero);
            if (i >= arrayLength) {
                System.out.println("\nFirst step: ");
                ge.display_matrix(array);
                int thirdZero = array[2][1];
                for (int k = 1; k < array[arrayLength - 1].length; k++) {
                    array[2][k] = array[2][k] - ((array[1][k] / array[1][1]) * thirdZero);
                }
                System.out.println("\nSecond step: ");
                ge.display_matrix(array);
            }
        }

        

    }

    public void display_matrix(int[][] array) {
        for (int[] array1 : array) {
            for (int j = 0; j < 4; j++) {
                System.out.print(array1[j] + "  \t");
            }
            System.out.println("");
        }
    }

}
